push-sdk-swift/AGPushAnalytics.swift

```diff
@@ -0,0 +1,130 @@
+                    sendMetrics(metrics, completionHandler: completionHandler)
+                }
+            }
+        }
+    }
+    
+    /**
+    * Send metrics to the AeroGear Push server when the app is first launched or bring from background to
+    * foreground due to a push notification.
+    *
+    * @param messageId The identifier of this push notification.
+    *
+    * @param completionHandler A block object to be executed when the send metrics operation finishes.
+    * Defaulted to no action.
+    */
+    class public func sendMetricsWhenAppAwoken(applicationState: UIApplicationState, userInfo: [NSObject:AnyObject], completionHandler: ((error: NSError? ) -> Void) = {(error: NSError?) in }) {
+        if applicationState == .Inactive || applicationState == .Background  {
+            //opened from a push notification when the app was on background
+            if let messageId = userInfo["aerogear-push-id"] as? String {
+                sendMetrics(messageId, completionHandler: completionHandler)
+            }
+        }
+    }
+    
+    /**
+    * Send metrics to the AeroGear Push server when the app is first launched or bring from background to
+    * foreground due to a push notification.
+    *
+    * @param messageId The identifier of this push notification.
+    *
+    * @param completionHandler A block object to be executed when the send metrics operation finishes.
+    * Defaulted to no action.
+    */
+    class private func sendMetrics(messageId: String, completionHandler: ((error: NSError? ) -> Void) = {(error: NSError?) in }) {
+        let variantId = NSUserDefaults.standardUserDefaults().valueForKey("variantID") as? String
+        let variantSecret = NSUserDefaults.standardUserDefaults().valueForKey("variantSecret") as? String
+        let urlString = NSUserDefaults.standardUserDefaults().valueForKey("serverURL") as? String
+        
+        
+        if let variantId = variantId, let variantSecret = variantSecret, let urlString = urlString {
+            let serverURL = NSURL(string: urlString)
+            // set up our request
+            let request = NSMutableURLRequest(URL: serverURL!.URLByAppendingPathComponent("rest/registry/device/pushMessage/\(messageId)"))
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            request.HTTPMethod = "PUT"
+            
+            // apply HTTP Basic
+            let basicAuthCredentials: NSData! = "\(variantId):\(variantSecret)".dataUsingEncoding(NSUTF8StringEncoding)
+            let base64Encoded = basicAuthCredentials.base64EncodedStringWithOptions(NSDataBase64EncodingOptions(0))
+            
+            request.setValue("Basic \(base64Encoded)", forHTTPHeaderField: "Authorization")
+            
+            let sessionConfig = NSURLSessionConfiguration.defaultSessionConfiguration()
+            let session = NSURLSession(configuration: sessionConfig)
```

Comment:
for consistency with registration let's add the callbacks to be called on the main queue e.g passing the delegateQueue parameter in the construction of the NSURLSession, [as in here](https://github.com/aerogear/aerogear-ios-push/blob/master/push-sdk-swift/AGDeviceRegistration.swift#L47):